console/src/state/mod.rs

```diff
@@ -39,15 +42,15 @@ enum Temporality {
 
 #[derive(Debug)]
 pub(crate) struct Metadata {
-    field_names: Vec<Arc<str>>,
-    target: Arc<str>,
+    field_names: Vec<InternedStr>,
```

Comment:
Since the interned strings are never pruned, this does mean that if a program had an event or span with a dynamically changing field name (is that a thing in `tracing`?), we could "leak" memory here... :shrug:

Reply:
`tracing` field names will never change, although the wire proto might allow changing them.

I considered pruning interned strings by storing `Weak` refs in the hashmap, but I decided we should do that in a follow-up branch. Could add it now if you prefer.

Reply:
If that can't happen in `tracing`, that's fine. We just need to be careful not to use this for a field "value" or similar strings. :)